tests/components/homematicip_cloud/fakeserver_tests/helper.py

```diff
@@ -0,0 +1,109 @@
+"""Helper for HomematicIP Cloud Tests."""
+import json
+from pathlib import Path
+
+from homematicip.base.base_connection import (
+    ATTR_CLIENT_AUTH,
+    BaseConnection,
+    HmipWrongHttpStatusError,
+)
+from homematicip_demo.fake_cloud_server import FakeCloudServer
+from werkzeug.wrappers import Request, Response
+
+
+def get_and_check_device_basics(hass, device_id, device_name, device_model):
+    """Get and test basic device."""
+
+    device = hass.states.get(device_id)
+    assert device is not None
+    assert device.attributes["model_type"] == device_model
+    assert device.name == device_name
+    return device
+
+
+class AsyncConnectionLocal(BaseConnection):
```

Comment:
I'm not sure this is the right approach. We're not interested in how the library connects to the cloud. We just want the library interfaces, facing home assistant, to work correctly and that we can use them in tests without that resulting in I/O.

If I understand correctly, we're now mocking the cloud part, keeping the library intact except for the cloud connection part. But we could instead just mock out the library so that it doesn't connect at all to a (fake) cloud.

Reply:
This approach removes I/O from HA testing.

With this approach i try to avoid the duplication of existing code for testing, because most part of the code is maintained in the upstream lib. 

I can move AsyncConnectionLocal to the upstream lib to get rid of this functionality from HA.

Reply:
I think we should mock the library and we should spec the mock to keep it simple on our end, with less work to implement the correct interfaces, while also staying future safe.

If we use the library object as spec a change in the library will propagate to the spec and the mock, so we make sure we're still using the correct interfaces.

Some instance attributes will usually need to be mocked specifically outside the spec, but instance methods will be mocked automatically.

Reply:
OK.
i close this PR and try to follow your suggestions in a new PR.
Thanks for feedback.